### DISCUSSION TITLE
JavaScript querySelector fails if you try to include/exclude classes with important prefix

### DISCUSSION THREAD

**Discussion by danieltamargo** [tailwindlabs/tailwindcss#9782] [Help] [answered]
Hello!

I've noticed this problem or bug, if you're using `document.querySelector()` or `document.querySelectorAll()` methods and you want to indicate in the selector the usage of any class that uses '!' as a prefix, it will fail and give you an error.

### Fast example:
Creating 2 similar blocks, the second one is using the important prefix '!'
![image](https://user-images.githubusercontent.com/32647764/200828210-0972862f-1519-4fff-9032-2676f69e547d.png)

Trying to select them
![image](https://user-images.githubusercontent.com/32647764/200828378-a37bcbc7-5a72-4aab-8058-8ba7d1b6cacc.png)

Is there any way to handle this other than avoiding using '!' prefix? Maybe I'm missing something

**Comment by wongjn** [comment, marked as answer]
You can try escaping the special character with double-backslash (`\\`):
```js
document.querySelectorAll('.test.\\!bg-red-400');
```

**Reply by danieltamargo** [reply]
This worked! 

Before posting this I tried to escape with **just one backslash**, that was my fault.

Gonna add this to my personal documentation so I don't forget about it.
